### PR TITLE
SAK-43102 - Extract AnnouncementWrapper

### PR DESCRIPTION
**Pull request by botimer**
This is a first step toward cleaning up AnnouncementAction. It extracts
the AnnouncementWrapper nested class, tidies up the handling of a couple
of entangled constants, removes a few references to TimeService, and
adds a couple of minimal tests to get things moving.

### CODE REVIEW

**Comment by steveswinsburg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 7
           

Complexity increasing per file
==============================
- announcement/announcement-tool/tool/src/test/org/sakaiproject/announcement/tool/AnnouncementWrapperTest.java  1
- announcement/announcement-tool/tool/src/java/org/sakaiproject/announcement/tool/AnnouncementWrapper.java  9
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/sakaiproject/sakai/pullRequest?prid=4915594&bid=16269484)

File: announcement/announcement-tool/tool/pom.xml [hunk 1]
```diff
@@ -75,6 +75,21 @@
             <groupId>org.springframework</groupId>
             <artifactId>spring-beans</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-core</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.powermock</groupId>
```
Comment (ern) [inline announcement/announcement-tool/tool/pom.xml]: Doesn't look like powermock dependencies are needed?

File: announcement/announcement-tool/tool/src/java/org/sakaiproject/announcement/tool/AnnouncementWrapper.java [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+import org.sakaiproject.user.api.ContextualUserDisplayService;
+import org.sakaiproject.user.api.User;
+import org.sakaiproject.util.api.FormattedText;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+
+/**
+ * Decorator for the "Message" class. It adds various properties to the decorated real Announcement message.
+ */
+public class AnnouncementWrapper implements AnnouncementMessage
```
Comment (ern) [inline announcement/announcement-tool/tool/src/java/org/sakaiproject/announcement/tool/AnnouncementWrapper.java]: Could use lombok to reduce boiler plate?

**Comment by botimer** [comment]
Yup... totally agreed that we could improve the code. I have no problem with creating new subtasks and doing that. Please see SAK-43101 for my overall philosophy here.

In particular, I want to do extractions minimally (only changing what is needed to disentangle), merge to master, and then follow up with incremental improvements. This is to preserve as much comparability as possible on the initial moves/renames and reduce the conceptual density of those particularly disruptive commits.

With respect to powermock, it is extraneous for this change; good eye. I expect there will be some interim steps where we have to do some unsavory fiddling soon but I should not have added it yet. But I guess I'd rather leave it in for now and pull it out if we find it unused when 43101 wraps up.

With respect to Lombok and boilerplate like getters and setters, those are personally lower priority than the other issues and smells. Problems like huge methods, complex state mutation in an undifferentiated hash, and going from enormous classes with dozens of dependencies to many more classes with few depenencies are my immediate focus. As I mentioned above, I would totally welcome someone to trim boilerplate, but I would like to see it done in small, discrete steps so there is clear history of the transformation and steady improvement in master.

**Pull request merged**